### PR TITLE
ZKUI-62: Add edit button for default retention at bucket level

### PR DESCRIPTION
**Pull request by ChengYanJin**
Retention mode: COMPLIANCE
We are not allow to switch from COMPLIANCE to GOVERNANCE from the UI
image
image

Retention mode: GOVERNANCE
image

### CODE REVIEW

File: src/react/databrowser/objects/ObjectLockSetting.jsx [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+const objectLockRetentionSettingsValidationRules = {
+  isRetentionEnabled: Joi.boolean().default(false),
+  retentionMode: Joi.when('isRetentionEnabled', {
+    is: Joi.equal(true),
+    then: Joi.string().required(),
+    otherwise: Joi.valid(),
+  }),
+  retentionUntilDate: Joi.date()
+    .format('YYYY-MM-DD')
+    .min('now')
```
Comment (JBWatenbergScality) [inline src/react/databrowser/objects/ObjectLockSetting.jsx:40]: nit: The minimum should be now or the current value

File: src/react/databrowser/objects/ObjectLockSetting.jsx [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    defaultRetentionUntilDate,
+  } = getDefaultRetention(objectMetadata);
+
+  useEffect(() => {
+    dispatch(getObjectMetadata(bucketNameParam, objectKey, versionId));
+  }, [dispatch, bucketNameParam, objectKey, versionId]);
+
+  const useFormMethods = useForm({
+    mode: 'all',
+    resolver: joiResolver(schema),
```
Comment (JBWatenbergScality) [inline src/react/databrowser/objects/ObjectLockSetting.jsx:80]: I think that to address my above comment you can likely passin the minimum date here

File: src/react/databrowser/objects/utils.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    .plus(Duration.fromObject({ days: 1 }))
+    .toISODate();
+
+  if (!d) {
+    return futureDate;
+  }
+
+  const previousRetainUntilDate = DateTime.fromSQL(d).toISODate();
+  return previousRetainUntilDate.valueOf() > futureDate.valueOf()
+    ? previousRetainUntilDate
```
Comment (JBWatenbergScality) [inline src/react/databrowser/objects/utils.js]: I think we should add one day here

**Review by JBWatenbergScality — APPROVED** [review]

**Review by gaudiauj — APPROVED** [review]

**Pull request merged**